lib/puppet/pops/evaluator/literal_evaluator.rb

```diff
@@ -67,6 +66,16 @@ def literal_LiteralRegularExpression(o)
     o.value
   end
 
+  def literal_QualifiedReference(o)
+    o.value
+  end
+
+  def literal_AccessExpression(o)
+    # to prevent parameters with [[]] like Optional[[String]]
+    throw :not_literal if o.keys.size == 1 && o.keys[0].is_a?(Model::LiteralList)
```

Comment:
I started to wonder if this `throw` might reject a valid type specification. For example, in the puppet language, `[1, 2]` is a literal array, so could it be that `Optional[[Integer]]` is actually valid? After more review, I don't think it is. If you want to accept an optional array of integers, I believe you have to say `Optional[Array[Integer]]`. So I think this check is ok, but I wanted to mention it.